point

```diff
@@ -87,17 +81,38 @@ program
 // program.option('--rpc <method> [params]', 'send a command to the daemon'); // todo
 program.parse(process.argv);
 
-// todo: mix other param arguments into config, like here:
-// let argv;
-// let config = rc('kadence', options(program.datadir), argv);
+const ctx = {};
 
-if (!program.config) {
-    program.config = path.join(program.datadir, 'config.json');
+if (program.datadir) {
+    process.env.DATADIR = program.datadir;
 }
 
-const ctx = {};
+function getContractAddress(name) {
+    const filename = path.resolve(__dirname, 'truffle', 'build', 'contracts', `${name}.json`);
+
+    if (!existsSync(filename)) {
+        return;
+    }
+
+    const {networks} = require(filename);
+
+    for (const network in networks) {
+        const {address} = networks[network];
+        if (address && typeof address === 'string') {
+            return address;
+        }
+    }
+}
+
+if (process.env.MODE === 'e2e') {
+    process.env.IDENTITY_CONTRACT_ADDRESS = getContractAddress('Identity');
+    process.env.STORAGE_PROVIDER_REGISTRY_CONTRACT_ADDRESS = getContractAddress('StorageProviderRegistry');
+}
```

Comment:
I think you only tested the app in the `dev` mode where you have the `truffle/` dir mounted. I bet it won't work in the `production` mode. From what I can see, you forgot to compile the `Identity` contract and store its' ABI in the `/truffle/build/contracts` dir. The app will not work without it. Please see the `compilePointContracts` in the deleted `patch-config.js` file. The function should run in all environments except the `e2e` one.

Please restore the function and add it along with the `getContractAddress` one to the `web3bridge` or a separate module importing it here.

Reply:
This function is not needed since the contracts are compiled by `deployer` in e2e mode. And you mentioned that it's only used in development?
I also tried running app in production and it works. Or did I get something wrong?

Reply:
> This function is not needed since the contracts are compiled by `deployer` in e2e mode. And you mentioned that it's only used in development?
The contract is compiled by the deployer only in the `e2e` mode. In all other modes we need to compile it or get it from a remote location which is not implemented at this point.
> I also tried running app in production and it works. Or did I get something wrong?
Ok then this could be because you've built an image with the `truffle/build/contracts` folder copied into container, which is wrong. Please add this folder to `.dockerignore`. Also mind it is already ignored by git, and that's for a purpose.